### PR TITLE
Issue/3207 add attributes part1

### PR DESCRIPTION
**Pull request by nbradbury**
Partially addresses #3207 by adding the "Add attributes" screen and populates it with both global attributes and the product's local attributes. 

This PR is fairly large, but is also missing a number of things, including the ability to actually add an attribute to the product. Likewise, UI touches such as loading indicators and skeletons will come later. This feature is only enabled for `DEBUG` builds, so it's fine to merge it incomplete into `develop`.

This PR also introduces a `CombinedAttributeModel` which can be instantiated from either a local (product-based) attribute or a global (store-wide) attribute.

add-attribute

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/Product.kt [hunk 1]
```diff
@@ -98,7 +98,9 @@ data class Product(
         val name: String,
         val options: List<String>,
         val isVisible: Boolean
-    ) : Parcelable
+    ) : Parcelable {
+        fun isLocalAttribute() = id == 0L
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/Product.kt]:
```suggestion
        val isLocalAttribute: Boolean
            get() = id == 0L
```
NP: It's more idiomatic in Kotlin to use a property for simple values like this.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt [hunk 2]
```diff
@@ -959,24 +969,50 @@ class ProductDetailViewModel @AssistedInject constructor(
...
     fun loadProductDraftAttributes() {
-        _attributeList.value =
-            viewState.productDraft?.let {
-                it.attributes
-            } ?: emptyList()
+        _attributeList.value = getProductDraftAttributes()
+    }
+
+    fun getProductDraftAttributes(): List<Product.Attribute> {
+        return viewState.productDraft?.let {
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt]: NP: No need to use `let`, we can just write `return viewState.productDraft?.attributes ?: emptyList()`

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/CombinedAttributeModel.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    override fun equals(other: Any?): Boolean {
+        return (other as? CombinedAttributeModel)?.let {
+            id == it.id &&
+                name == it.name
+        } ?: false
+    }
+
+    override fun hashCode() = super.hashCode()
+
+    fun isGlobalAttribute() = id != 0L
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/CombinedAttributeModel.kt]: NP: Sane as the product attribute.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/CombinedAttributeModel.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package com.woocommerce.android.ui.products.variations.attributes
+
+import com.woocommerce.android.model.Product
+import com.woocommerce.android.model.ProductGlobalAttribute
+
+/**
+ * A "combined" product attribute is one that can be created from either a local (product-based) attribute
+ * or a global (store-based) attribute
+ */
+class CombinedAttributeModel(
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/CombinedAttributeModel.kt]: Why not make it a `data class`? You'd get the `equals()` and `hashCode()` for free.

**Review by 0nko — COMMENTED** [review]
It's looking good! :+1: I just left some very minor nit pick comments in the code.

**Comment by nbradbury** [comment]
Thanks, @0nko. I've addressed the issues you raised.

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**